src/v/kafka/server/connection_context.cc

```diff
@@ -311,6 +311,8 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
               const auto correlation = rctx.header().correlation;
               const sequence_id seq = _seq_idx;
               _seq_idx = _seq_idx + sequence_id(1);
+              auto record_time_quota = _proto.quota_mgr().time_quota_recorder(
```

Comment:
Is it possible for another client request (from this client, or any other client) to come in while we're still processing this request?

Reply:
Checkout `kafka/protocol.cc` pretty much there is a run loop that processes requests one at a time. However there is a caveat, within this file there is an out of band background loop, so any code that is placed there does not have the same guarantees.